### PR TITLE
Fixes tests on relay modern branch

### PR DESCRIPTION
**Pull request by l2succes**
All the tests (except for one that still needs to be fixed) are now passing.
screen shot 2017-09-06 at 8 57 35 am 1

Remaining work

- [x] Fix storybooks

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Running your Dangerfile has Failed
  </td>
    </tr>
  </tbody>
</table>




## Danger has errored

Error: TypeError

```sh
TypeError: Cannot read property '' of undefined
    at ModuleMap._getModuleMetadata (/home/travis/build/artsy/emission/node_modules/jest-haste-map/build/module_map.js:84:30)
    at ModuleMap.getModule (/home/travis/build/artsy/emission/node_modules/jest-haste-map/build/module_map.js:40:25)
    at Resolver.getModule (/home/travis/build/artsy/emission/node_modules/jest-runtime/node_modules/jest-resolve/build/index.js:191:28)
    at Resolver._isModuleResolved (/home/travis/build/artsy/emission/node_modules/jest-runtime/node_modules/jest-resolve/build/index.js:300:20)
    at Resolver._getAbsolutPath (/home/travis/build/artsy/emission/node_modules/jest-runtime/node_modules/jest-resolve/build/index.js:277:17)
    at Resolver.getModuleID (/home/travis/build/artsy/emission/node_modules/jest-runtime/node_modules/jest-resolve/build/index.js:252:31)
    at Runtime.requireModule (/home/travis/build/artsy/emission/node_modules/jest-runtime/build/index.js:276:37)
    at /home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:133:33
    at ensureCleanDangerfile (/home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:173:5)
    at Object.<anonymous> (/home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:132:21)
```


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: __mocks__/react-relay.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 // Comes from https://github.com/facebook/relay/issues/161
 // Should be used in tests as a replacement for `react-storybooks-relay-container`
 //
-var Relay = require.requireActual('react-relay')
+var Relay = require.requireActual("react-relay")
 
 module.exports = {
   QL: Relay.QL,
   Mutation: Relay.Mutation,
   Route: Relay.Route,
```
Comment (alloy) [inline __mocks__/react-relay.js]: These are no longer needed are they?

**Comment by alloy** [comment]
Looks great 👍 Go ahead and merge once done/green.

File: __mocks__/react-relay.js (hunk 1 above)
Comment (l2succes) [reply __mocks__/react-relay.js]: I didn't check if there were any references left but if there aren't, it's safe to remove. I'll check now

**Comment by orta** [comment]
I'm gonna try get this green for the next 30m

**Comment by orta** [comment]
Bumped jest-snapshots-svg to support Jest 21 types ( https://github.com/orta/jest-snapshots-svg/commit/1148a3c34d3f9c17a218d0d9b2053a985415b5ca ) and updated jest-styled-components.

**Comment by orta** [comment]
Got pretty far, but I'm not 100% that I'm in a direction that I totally understand. The next step for someone is to do: 

```sh
yarn remove @types/react-test-renderer
yarn upgrade @types/react @types/react-native
```

This opens up ~80 errors, related to our use of `children` inside components and references to `global`.

```
154     if (__DEV__ && (global as any).originalXMLHttpRequest !== undefined) {
                        ~~~~~~

src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx(154,21): error TS2304: Cannot find name 'global'.


114     <Button onPress={props.goToArtist}>
                ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/lib/Components/Consignments/Components/ArtworkConsignmentTodo.tsx(114,13): error TS2322: Type '{ onPress: () => void; children: Element[]; }' is not assignable to type 'IntrinsicAttributes & TouchableHighlightProperties'.

```

File: src/lib/Containers/Gene.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as _ from "lodash"
 import * as React from "react"
 import ParallaxScrollView from "react-native-parallax-scroll-view"
-import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
+import { createRefetchContainer, graphql, RelayProp, RelayRefetchProp } from "react-relay"
```
Comment (alloy) [inline src/lib/Containers/Gene.tsx]: Why is this needed?

**Comment by alloy** [comment]
The references to the `global` variable can probably be fixed by just removing those references. E.g. just reference `originalXMLHttpRequest`.

**Comment by l2succes** [comment]
I got `yarn ci` to pass successfully on my local machine but I'm struggling to get it working on travis CI. So I'll merge this branch into `relay-modern` and see if it works there.
